src/components/SelectV2/Select.tsx

```diff
@@ -0,0 +1,201 @@
+import { DesktopDropdown } from './DesktopDropdown'
+
+interface AdornmentProps {
+  $start?: boolean
+}
+
+const Adornment = styled.span<AdornmentProps>`
+  display: grid;
+  place-items: center;
+  pointer-events: none;
+  ${props =>
+    Boolean(props.$start)
+      ? 'grid-area: 1 / 1 / 1 / 2'
+      : 'grid-area: 1 / -2 / 1 / -1'}
+`
+
+interface ContainerProps {
+  hasStartAdornment: boolean
+}
+
+const Container = styled.span<ContainerProps>`
+  display: grid;
+  grid-template-columns: 1fr ${space[48]};
+  position: relative;
+  isolation: isolate;
+
+  ${props =>
+    props.hasStartAdornment &&
+    css`
+      grid-template-columns: ${space[48]} 1fr ${space[48]};
+    `};
+`
+
+interface StyledSelectProps {
+  variant: SelectVariant
+}
+
+const StyledSelect = styled.select<StyledSelectProps>`
+  grid-area: 1 / 1 / 1 / 4;
+  color: transparent;
+  appearance: none;
+  background-color: transparent;
+
+  &:focus {
+    outline: none;
+  }
+
+  @media (hover: hover) {
+    display: none;
+    pointer-events: none;
+  }
+
+  ${props =>
+    props.variant === 'short' &&
+    css`
+      position: absolute;
+      inset: 0;
+      z-index: -1;
+    `}
+`
+
+interface DisplayValueProps {
+  disabled?: boolean
+  hasStartAdornment: boolean
+  rounded: boolean
+}
+
+const DisplayValue = styled.span<DisplayValueProps>`
+  grid-area: 1 / 1 / 1 / 4;
+  font-size: ${fontSize[18]};
+  line-height: ${space[56]};
+  height: ${space[56]};
+  padding-inline: ${space[16]} ${space[48]};
+  color: ${color.foreground};
+  display: grid;
+  grid-template-columns: auto 1fr;
+  align-items: center;
+  justify-content: start;
+  gap: ${space[8]};
+  border-radius: ${props => (props.rounded ? space[32] : radius.md)};
+  background-color: ${color.elevatedBackground};
+  pointer-events: none;
+
+  ${props =>
+    props.hasStartAdornment &&
+    css`
+      padding-inline-start: ${space[48]};
+    `}
+
+  ${Container}:focus-within & {
+    box-shadow: ${shadow.strong};
+    background-color: ${color.background};
+  }
+
+  ${props =>
+    props.disabled &&
+    css`
+      background-color: ${color.inactiveBackground};
+      cursor: not-allowed;
+      pointer-events: auto;
+    `}
+`
+
+export interface OptionProps extends OptionHTMLAttributes<HTMLOptionElement> {
+  startAdornment?: ReactNode
+}
+
+export type SelectVariant = 'default' | 'short' | 'long' | 'shortvalue'
+
+export interface SelectProps extends InputHTMLAttributes<HTMLSelectElement> {
+  options: OptionProps[]
+  'aria-label': string
+  startAdornment?: ReactNode
+  rounded?: boolean
+  variant?: SelectVariant
+}
+
+export const Select = ({
+  options,
+  startAdornment = null,
+  rounded = false,
+  variant = 'default',
+  onChange,
+  'aria-label': ariaLabel,
+  ...props
+}: SelectProps) => {
+  const selectboxRef = useRef<HTMLSelectElement>(null)
+  const [currentValue, setCurrentValue] = useState(
+    props.value || props.defaultValue || options[0].value
```

Comment:
We can destruct `defaultValue` as well right? Since we're using other `value` variables in this file I can see why we don't want to destruct that, although that's also possible.